tests/test_templates.py

```diff
@@ -12,106 +12,351 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Unit tests for the integration of the methods from :mod:`pennylane.init` with the :mod:`pennylane.templates` module.
+Integration tests for templates, including integration with initialization functions
+in :mod:`pennylane.init`, running templates in larger circuits,
+combining templates, using positional and keyword arguments, and using different interfaces.
+
+New tests are added as follows:
+
+* When adding a new interface, try to import it and extend the fixture ``interfaces``. Add interface to
+  TestGradientIntegration tests.
+
+* When adding a new template, extend the fixtures ``qubit_const`` or ``cv_const`` by a *list* of arguments to the
+  template. Note: Even if the template takes only one argument, it has to be wrapped in a list (i.e. [weights]).
+
+* When adding a new parameter initialization function, extend the fixtures ``qubit_func`` or ``cv_func`` by the
+  function.
 """
 # pylint: disable=protected-access,cell-var-from-loop
 import pytest
+import numpy as np
 import pennylane as qml
-from pennylane.templates.layers import (CVNeuralNetLayers, CVNeuralNetLayer,
-                                        StronglyEntanglingLayers, StronglyEntanglingLayer,
-                                        RandomLayers, RandomLayer)
-from pennylane.init import (strong_ent_layers_uniform, strong_ent_layer_uniform,
-                            strong_ent_layers_normal, strong_ent_layer_normal,
-                            random_layers_uniform, random_layer_uniform,
-                            random_layers_normal, random_layer_normal,
-                            cvqnn_layers_uniform, cvqnn_layer_uniform,
-                            cvqnn_layers_normal, cvqnn_layer_normal)
+from pennylane.templates.layers import Interferometer
+from pennylane.templates.layers import (CVNeuralNetLayers,
+                                        StronglyEntanglingLayers,
+                                        RandomLayers)
+from pennylane.templates.embeddings import (AmplitudeEmbedding,
+                                            BasisEmbedding,
+                                            AngleEmbedding,
+                                            SqueezingEmbedding,
+                                            DisplacementEmbedding)
+from pennylane.init import (strong_ent_layers_uniform,
+                            strong_ent_layers_normal,
+                            random_layers_uniform,
+                            random_layers_normal,
+                            cvqnn_layers_uniform,
+                            cvqnn_layers_normal)
 
 
-class TestParameterIntegration:
-    """ Integration tests for the parameter generation methods from pennylane.init
-    and pennylane.templates.layers."""
+interfaces = [('numpy', np.array)]
 
```

Comment:
From here on the diff becomes unreadable...I recommend looking at it in the file...